dace/codegen/targets/intel_fpga.py

```diff
@@ -257,8 +257,6 @@ def make_read(self, defined_type, dtype, var_name, expr, index, is_pack,
         if defined_type == DefinedType.Stream:
             read_expr = "read_channel_intel({})".format(expr)
         elif defined_type == DefinedType.StreamArray:
-            # remove "[0]" index as this is not allowed if the subscripted value is not an array
-            expr = expr.replace("[0]", "")
```

Comment:
Is this handled differently?
IIRC there was a case where a single stream was accessed as an array.
(But this wasn't the right place to fix it)

Reply:
This was a bad hack that covered up the actual problem, which was that we defined memlets `#define pipe pipes[i]` as a `StreamArray`, when it should be a `Stream`. This is fixed now, so this hack is no longer necessary :-)